client/src/components/CheckoutForm.js

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from "react";
+import { CardElement, useStripe, useElements } from "@stripe/react-stripe-js";
+import {
+  Button,
+  Paper,
+  Grid,
+  Typography,
+  FormControl,
+  InputLabel,
+  Input,
+} from "@material-ui/core";
+import { useTheme, makeStyles } from "@material-ui/core/styles";
+
+const checkoutFormStyles = makeStyles((theme) => ({
+  paperContainer: {
+    width: "50%",
+    padding: "4rem",
+    background: ` ${theme.palette.secondary.main}`,
+    borderRadius: "18px",
+  },
+  buttonText: {
+    color: "white",
+    marginTop: "3rem",
+    padding: "0.5rem 1.5rem 0.5rem 1.5rem",
+  },
+  cardElementContainer: {
+    height: "45px",
+  },
+  cardContainer: {
+    border: `2px solid ${theme.palette.primary.main}`,
+    padding: "16px",
+    borderRadius: "8px",
+    marginTop: "3rem",
+  },
+  topSpacing: {
+    marginTop: "2rem",
+  },
+  inputLabel: {
+    color: "white",
+  },
+}));
+
+const CheckoutForm = () => {
+  const stripe = useStripe();
+  const elements = useElements();
+  const theme = useTheme();
+  const classes = checkoutFormStyles(theme);
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  };
+  const handleNameChange = (e) => {
+    setName(e.target.value);
+  };
+
+  const cardElementOptions = {
+    iconStyle: "solid",
+    style: {
+      base: {
+        iconColor: `${theme.palette.primary.main}`,
+        color: "white",
+        fontWeight: 500,
+        fontFamily: "Roboto, Open Sans, Segoe UI, sans-serif",
+        fontSize: "16px",
+        border: `1px solid ${theme.palette.primary.main}`,
+        fontSmoothing: "antialiased",
+        "::placeholder": { color: `white` },
+      },
+      invalid: {
+        iconColor: "red",
+        color: "red",
+      },
+    },
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
```

Comment:
same comment as above. You can try it without `preventDefault` and see if you get undesired behaviour

Reply:
Could deleting this possibly be what's causing the stripe functionality to stop working?